nms/app/packages/magmalte/app/views/equipment/GatewayDetailConfigEdit.js

```diff
@@ -822,3 +869,165 @@ export function RanEdit(props: Props) {
     </>
   );
 }
+
+export function ApnResourcesEdit(props: Props) {
+  const classes = useStyles();
+  const enqueueSnackbar = useEnqueueSnackbar();
+  const [error, setError] = useState('');
+  const ctx = useContext(GatewayContext);
+  const apnCtx = useContext(ApnContext);
+  const lteCtx = useContext(LteNetworkContext);
+  const [apnResources, _] = useState<apn_resources>(
+    props.gateway?.apn_resources || DEFAULT_GATEWAY_CONFIG.apn_resources,
+  );
+  const [apnResourcesRows, setApnResourcesRows] = useState(
+    Object.keys(apnResources).map(apn => apnResources[apn]),
+  );
+
+  const handleApnResourcesChange = (key: string, val, index: number) => {
+    const rows = apnResourcesRows;
+    rows[index][key] = val;
+    setApnResourcesRows([...rows]);
+  };
+  const deleteApn = deletedApn =>
+    setApnResourcesRows([
+      ...apnResourcesRows.filter(apn => apn !== deletedApn),
+    ]);
+
+  const addApnResource = () => {
+    setApnResourcesRows([
+      ...apnResourcesRows,
+      {apn_name: '', id: '', vlan_id: null},
+    ]);
+  };
+  const gatewayApnResources = {};
```

Comment:
minor nit: since it is not being used anywhere else,we could possibly move this into onSave block